desc/magnetic_fields/_core.py

```diff
@@ -2644,60 +2666,48 @@ def field_line_integrate(
     min_step_size = jnp.where(
         phis[-1] > phis[0], min_step_size, -jnp.abs(min_step_size)
     )
-
-    @jit
-    def odefun(s, rpz, args):
-        rpz = rpz.reshape((3, -1)).T
-        field = args[0]
-        r = rpz[:, 0]
-        br, bp, bz = (
-            scale
-            * field.compute_magnetic_field(
-                rpz, params, basis="rpz", source_grid=source_grid, chunk_size=chunk_size
-            ).T
-        )
-        return jnp.array(
-            [r * br / bp * jnp.sign(bp), jnp.sign(bp), r * bz / bp * jnp.sign(bp)]
-        ).squeeze()
+    max_steps = (
+        max_steps
+        if max_steps is not None
+        else int((phis[-1] - phis[0]) / min_step_size * 1000)
+    )
 
     # diffrax parameters
-
-    def default_terminating_event_fxn(state, **kwargs):
-        R_out = jnp.logical_or(state.y[0] < bounds_R[0], state.y[0] > bounds_R[1])
-        Z_out = jnp.logical_or(state.y[2] < bounds_Z[0], state.y[2] > bounds_Z[1])
+    def default_terminating_event(t, y, args, **kwargs):
```

Comment:
this is another locally defined function that could trigger recompilation, so should probably be moved outside this scope